CMakeLists.txt

```diff
@@ -9,82 +9,50 @@
 cmake_minimum_required(VERSION 3.1)
 project(xeus)
 
-# Dependencies
-# ============
-
-find_package(xtl REQUIRED)
-find_package(ZeroMQ REQUIRED)
-find_package(cppzmq REQUIRED)
-find_package(cryptopp REQUIRED)
-
-# Compilation flags
-# =================
-
-include(CheckCXXCompilerFlag)
-
-string(TOUPPER "${CMAKE_BUILD_TYPE}" U_CMAKE_BUILD_TYPE)
+set(XEUS_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/include)
+set(XEUS_SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src)
 
-OPTION(DISABLE_ARCH_NATIVE "disable -march=native flag" OFF)
+# Versionning
+# ===========
 
-if (CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Intel")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wunused-parameter -Wextra -Wreorder")
-    if (DISABLE_ARCH_NATIVE)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mtune=generic")
-    else()
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native")
+# Project version
+file(STRINGS "${XEUS_INCLUDE_DIR}/xeus/xeus.hpp" xeus_version_defines
+     REGEX "#define XEUS_VERSION_(MAJOR|MINOR|PATCH)")
+foreach(ver ${xeus_version_defines})
+    if(ver MATCHES "#define XEUS_VERSION_(MAJOR|MINOR|PATCH) +([^ ]+)$")
+        set(XEUS_VERSION_${CMAKE_MATCH_1} "${CMAKE_MATCH_2}" CACHE INTERNAL "")
     endif()
+endforeach()
+set(XEUS_VERSION
+    ${XEUS_VERSION_MAJOR}.${XEUS_VERSION_MINOR}.${XEUS_VERSION_PATCH})
+message(STATUS "xeus version: v${XEUS_VERSION}")
 
-    CHECK_CXX_COMPILER_FLAG("-std=c++11" HAS_CPP11_FLAG)
-
-    if (HAS_CPP11_FLAG)
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
-    else()
-        message(FATAL_ERROR "Unsupported compiler -- xeus requires C++11 support!")
+# Binary version
+# See the following URL for explanations about the binary versionning
+# https://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html#Updating-version-info
+file(STRINGS "${XEUS_INCLUDE_DIR}/xeus/xeus.hpp" xeus_version_defines
+    REGEX "#define XEUS_BINARY_(CURRENT|REVISION|AGE)")
+foreach(ver ${xeus_version_defines})
+    if(ver MATCHES "#define XEUS_BINARY_(CURRENT|REVISION|AGE) +([^ ]+)$")
+        set(XEUS_BINARY_${CMAKE_MATCH_1} "${CMAKE_MATCH_2}" CACHE INTERNAL "")
     endif()
+endforeach()
+set(XEUS_BINARY_VERSION
+    ${XEUS_BINARY_CURRENT}.${XEUS_BINARY_REVISION}.${XEUS_BINARY_AGE})
```

Comment:
I don't believe you've implemented this correctly. The 'current', 'revision', and 'age' defined by libtool are _not_ the 3 numbers in the soversion. See the implementation for Linux at https://git.savannah.gnu.org/cgit/libtool.git/tree/build-aux/ltmain.in#n7042

Reply:
Oh, I see you actually do the correct thing when setting the library properties. Bit confusing, but still correct, seems like.

Reply:
@QuLogic do you have a suggestion on a clearer implementation?

Reply:
@QuLogic Do you think an additional `XEUS_SO_VERSION` variable would help here?

Reply:
I'm not sure if a second variable would help, but maybe moving the calculation up here might lessen confusion?

Since `${XEUS_BINARY_CURRENT}.${XEUS_BINARY_REVISION}.${XEUS_BINARY_AGE}` is not truly what ends up used (on Linux), it seems weird to give it a name at all.